packages/axios-extension/src/abap/lrep-service.ts

```diff
@@ -0,0 +1,163 @@
+
+/**
+ * Returns the namespace as string.
+ *
+ * @param namespace either as string or as object
+ * @returns serialized namespace
+ */
+function getNamespaceAsString(namespace: Namespace): string {
+    return namespace['id'] ? `apps/${namespace['reference']}/appVariants/${namespace['id']}/` : (namespace as string);
+}
+
+/**
+ * A class respresenting the design time adaptation service allowing to deploy adaptation projects to an ABAP system.
+ */
+export abstract class LayeredRepositoryService extends Axios implements Service {
+    public static readonly PATH = '/sap/bc/lrep';
+
+    public log: Logger;
+
+    /**
+     * Check whether a variant with the given namespace already exists.
+     *
+     * @param namespace either as string or as object
+     * @returns the Axios response object for futher processing
+     */
+    public async check(namespace: Namespace): Promise<AxiosResponse> {
+        return this.get('/dta_folder/', {
+            params: {
+                name: getNamespaceAsString(namespace),
+                layer: 'CUSTOMER_BASE' as Layer
+            }
+        });
+    }
+
+    /**
+     * Fetch a csrf token required for deployment. The token will be returned as header 'x-csrf-token'.
+     *
+     * @returns the Axios response object for futher processing
+     */
+    public async getCsrfToken(): Promise<AxiosResponse> {
+        return this.get('/actions/getcsrftoken/', {
+            headers: {
+                'X-Csrf-Token': 'Fetch'
+            }
+        });
+    }
+
+    /**
+     * Deploy the given archive either by creating a new folder in the layered repository or updating an existing one.
+     *
+     * @param archivePath path to a zip archive containing the adaptation project
+     * @param config adataption project deployment configuration
+     * @returns the Axios response object for futher processing
+     */
+    public async deploy(archivePath: string, config: AdaptationConfig): Promise<AxiosResponse> {
+        const archive = readFileSync(archivePath);
+
+        const tokenResponse = await this.getCsrfToken();
+        const checkResponse = await this.check(config.namespace);
+        const params: object = {
+            name: getNamespaceAsString(config.namespace),
+            layer: 'CUSTOMER_BASE' as Layer
+        };
+        if (config.package) {
+            params['package'] = config.package;
+            if (config.package.toLowerCase() !== '$tmp') {
+                params['changeList'] = config.transport;
+            }
+        }
+        const response = await this.request({
+            method: checkResponse.status === 200 ? 'PUT' : 'POST',
+            url: '/dta_folder/',
+            data: archive,
+            params,
+            headers: {
+                'Content-Type': 'application/octet-stream',
+                'X-Csrf-Token': tokenResponse.headers['x-csrf-token']
+            }
+        });
+
+        const info = response.data ? JSON.parse(response.data) : {};
```

Comment:
`JSON.parse` can throw an error, may need to handle it

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/469/commits/b8ee2925a0889471a628b6eb9a0444917d1e9e83